### PR TITLE
use terminaltables3, migrate to pyproject.toml

### PR DESCRIPTION
**Pull request by branchv**
This switches `terminaltables` with `terminaltables3` since the current maintainer lost PyPI access and had to republish under the new name (see https://github.com/matthewdeanmartin/terminaltables3/issues/3#issuecomment-2220673820)

Also, I took the opportunity to migrate to `pyproject.toml`, which is the modern [replacement for setup.py](https://packaging.python.org/en/latest/discussions/setup-py-deprecated/)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[project]
+name = "keep"
+version = "2.10.1"
+description = "Personal shell command keeper"
+readme = "README.md"
+authors = [
+    { name = "Himanshu Mishra", email = "himanshu.mishra.kgp@gmail.com" },
+]
+dependencies = ["PyGithub", "click", "requests", "terminaltables3"]
```
Comment (rahuliyer95) [inline pyproject.toml:9]: We should pin the version for `PyGithub` to `1.4.7` as was the case in `requirements.txt`

**Review by rahuliyer95 — COMMENTED** [review]
LGTM! Will let @OrkoHunter take another pass and merge

File: pyproject.toml (hunk 1 above)
Comment (branchvincent) [reply pyproject.toml:9]: it's unpinned in `setup.py` though (which is what users see/install and I agree is the way to go): https://github.com/OrkoHunter/keep/blob/e6eadb926464af1ed41e9c8c69b72882b6bb1d77/setup.py#L28-L33

**Review by OrkoHunter — APPROVED** [review]
Thank you @branchvincent and Rahul!

**Comment by branchvincent** [comment]
thanks! any chance this will land in a release soon?

**Comment by OrkoHunter** [comment]
@branchvincent Released 2.11 ! 

https://pypi.org/manage/project/keep/releases/

**Comment by OrkoHunter** [comment]
@rahuliyer95 Can you share your pypi.org username with me? I'm happy to add you as a maintainer on pypi.org. Right now I'm the sole maintainer there and it doesn't look good. 😁

**Comment by rahuliyer95** [comment]
> @rahuliyer95 Can you share your pypi.org username with me? I'm happy to add you as a maintainer on pypi.org. Right now I'm the sole maintainer there and it doesn't look good. 😁

It's [rahuliyer95](https://pypi.org/user/rahuliyer95/)

**Comment by branchv** [comment]
awesome thank you @OrkoHunter!

**Pull request merged**